lib/spack/spack/cmd/buildcache.py

```diff
@@ -388,7 +388,9 @@ def _createtarball(env, spec_yaml=None, packages=None, add_spec=True,
                 if d == 0:
                     continue
 
-                if node.external or node.virtual:
+                lookup = spack.store.db.query_one(node)
+
+                if node.external or node.virtual or lookup is None:
                     tty.debug('skipping external or virtual dependency %s' %
```

Comment:
This message now needs to include "not installed" as a reason we could reach this block of code.